scripts/lib/CIME/case/check_input_data.py

```diff
@@ -6,9 +6,45 @@
 from CIME.XML.inputdata import Inputdata
 import CIME.Servers
 
-import glob
+import glob, hashlib
 
 logger = logging.getLogger(__name__)
+# The inputdata_checksum.dat file will be read into this hash if it's available
+chksum_hash = dict()
+
+
+def _download_checksum_file(server, input_data_root):
+    """
+    Return True if successfully downloaded
+    """
+    rel_path = "../inputdata_checksum.dat"
+    full_path = os.path.join(input_data_root, "inputdata_checksum.dat")
+    protocol = type(server).__name__
+    if protocol == 'svn':
+        # svn server does not have this file
+        return False
+    logging.info("Trying to download file: '{}' to path '{}' using {} protocal.".format(rel_path, full_path, protocol))
+    tmpfile = None
+    if os.path.isfile(full_path):
+        tmpfile = full_path+".tmp"
+        os.rename(full_path, tmpfile)
+    # Use umask to make sure files are group read/writable. As long as parent directories
+    # have +s, then everything should work.
+    with SharedArea():
+        success = server.getfile(rel_path, full_path)
+        if success:
+            if tmpfile:
+                os.remove(tmpfile)
+        else:
+            if os.path.isfile(tmpfile):
+                os.rename(tmpfile, full_path)
+                logger.warning("Could not automatically download file "+full_path+
+                           " Restoring existing version.")
+            else:
+                logger.warning("Could not automatically download file "+full_path+
+                           "download from ftp:ftp.cgd.ucar.edu/cesm/inputdata_chksum.dat")
```

Comment:
Shouldn't this hard-coded path be replaced with the actual remote path for this particular server?